albatross/core/prediction.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2018 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef ALBATROSS_CORE_PREDICTION_H
+#define ALBATROSS_CORE_PREDICTION_H
+
+namespace albatross {
+
+// This is effectively just a container that allows us to develop methods
+// which behave different conditional on the type of predictions desired.
+template <typename T> struct PredictTypeIdentity { typedef T type; };
```

Comment:
So this is effectively used as a tag type? If that's the case you don't strictly need to be declaring it as a reference in the function signatures where it is used. Empty classes like this take the minimum amount of space (usually a byte) and it's the same cost to copy the object as it is to copy a reference to it.

Reply:
Right, it's just a tag type. So would you suggest switching to a signature for predict that looks like:

```
  template <typename FeatureType, typename FitType, typename PredictType>
  PredictType predict(const std::vector<FeatureType> &features,
                                  const FitType &least_squares_fit,
                                  PredictTypeIdentity<PredictType>) const
```
Is it worth throwing `const` in there even though there's nothing to modify?